jaxrs/src/main/java/org/jboss/as/jaxrs/JaxrsContextParamHandler.java

```diff
@@ -0,0 +1,271 @@
+   private static AttributeDefinition getAttributeDefinition(String name, ModelType modelType, ModelNode defaultValue) {
+      if (NO_DEFAULT.equals(defaultValue.asString())) {
+         return new SimpleAttributeDefinitionBuilder(name, modelType)
+               .setRequired(false)
+               .setAllowExpression(true)
+               .build();
+      } else {
+         return new SimpleAttributeDefinitionBuilder(name, modelType)
+               .setRequired(false)
+               .setAllowExpression(true)
+               .setDefaultValue(defaultValue)
+               .build();
+      }
+   }
+
+   private static AttributeDefinition getListAttributeDefinition(String name, ModelType modelType) {
+      return new StringListAttributeDefinition.Builder(name)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setValidator(JavaIdListValidator.INSTANCE)
+            .build();
+   }
+
+   private static AttributeDefinition getMapAttributeDefinition(String name, ModelType modelType) {
+      return new StringListAttributeDefinition.Builder(name)
+            .setRequired(false)
+            .setAllowExpression(true)
+            .setValidator(JavaMapValidator.INSTANCE)
+            .build();
+   }
+
+   private static void registerAttribute(ManagementResourceRegistration resourceRegistration, AttributeDefinition attributeDefinition) {
+      resourceRegistration.registerReadWriteAttribute(attributeDefinition, new ContextParamReadHandler(), new ContextParamWriteHandler(attributeDefinition));
+   }
+
+   ///////////////////////////////////////////////////////////////////////////////
+   ///                               Handler Classes                           ///
+   ///////////////////////////////////////////////////////////////////////////////
+   static class ContextParamWriteHandler extends AbstractWriteAttributeHandler<Void> {
+
+      private AttributeDefinition attributeDefinition;
+
+      private ContextParamWriteHandler(AttributeDefinition attributeDefinition) {
+         super(attributeDefinition);
+         this.attributeDefinition = attributeDefinition;
+      }
+
+      @Override
+      protected boolean applyUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName,
+            ModelNode resolvedValue, ModelNode currentValue, HandbackHolder<Void> handbackHolder) throws OperationFailedException {
+         JaxrsContextParamHandler.setContextParam(attributeName, resolvedValue);
+         return false;
+      }
+
+      @Override
+      protected void revertUpdateToRuntime(OperationContext context, ModelNode operation, String attributeName, ModelNode valueToRestore, ModelNode valueToRevert, Void handback) throws OperationFailedException{
+         JaxrsContextParamHandler.setContextParam(attributeName, valueToRestore);
+      }
+   }
+
+   static class ContextParamReadHandler extends ReadAttributeHandler {
```

Comment:
Delete this class, it seems to be the root of my observation regarding the defaults. In the call to register the attribute, pass in a null handler and it will add the proper read handler which deals with the defaults for you.